pkg/p2p/kadcast/packet.go

```diff
@@ -0,0 +1,136 @@
+package kadcast
+
+import (
+	"encoding/binary"
+	"log"
+	"net"
+)
+
+type Packet struct {
+	headers [24]byte
+	payload []byte
+}
+
+func makePacket(headers [24]byte, payload []byte) Packet {
+	return Packet {
+		headers: headers,
+		payload: payload,
+	}
+}
+
+// Gets a stream of bytes and slices it between headers of Kadcast
+// protocol and the payload.
+func getPacketFromStream(stream []byte) Packet {
+	var headers [24]byte
+	copy(headers[:], stream[0:23])
+	return Packet{
+		headers: headers,
+		payload: stream[23:],
+	}
+}
+
+// Deserializes the packet into an slice of bytes.
+func (pac Packet) asBytes() []byte {
+	return append(pac.headers[:], pac.payload[:]...)
+}
+
+// Returns the headers info sliced into three pieces:
+// Packet type, SenderId, IdNonce and senderPort.
+func (pac Packet) getHeadersInfo() (byte, [16]byte, [4]byte, [2]byte) {
+	// Construct type, senderID and Nonce
+	typ := pac.headers[0]
+	var senderID [16]byte
+	copy(senderID[:], pac.headers[1:17])
+	var nonce [4]byte
+	copy(nonce[:], pac.headers[17:21])
+	var peerPort [2]byte
+	copy(peerPort[:], pac.headers[21:23])
+	return typ, senderID, nonce, peerPort
+}
+
+// Gets the Packet headers parts and puts them into the
+// header attribute of the Packet.
+func (pack *Packet) setHeadersInfo(tipus byte, router Router, destPeer Peer) {
+	var headers []byte
```

Comment:
Since you already know the size of this slice beforehand (24 bytes) you could do something like this
```
headers := make([]byte, 0, 24)
```

Which saves you the allocations made when you are appending stuff, since you don't have the grow the slice on the fly.

Reply:
Will modify it on the final commits! Thanks, didn't saw it!